spec/rubocop/cop/rspec/empty_line_after_hook_spec.rb

```diff
@@ -19,6 +19,126 @@
     RUBY
   end
 
+  it 'allows multiple one-liner blocks' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        before { do_something }
+        before { do_something_else }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Add an empty line after `before`.
+        it { does_something }
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        before { do_something }
+        before { do_something_else }
+
+        it { does_something }
+      end
+    RUBY
+  end
+
+  it 'does not register an offense for chainer one-liner `before` hooks' do
+    expect_no_offenses(<<-RUBY)
+      RSpec.describe User do
+        before { do_something }
+        before { do_something_else }
+
+        it { does_something }
+      end
+    RUBY
+  end
+
+  it 'registers an offense for multiline blocks without empty line after' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        before do
+          do_something
+        end
+        ^^^ Add an empty line after `before`.
+        before { do_something_else }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Add an empty line after `before`.
+        it { does_something }
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        before do
+          do_something
+        end
+
+        before { do_something_else }
+
+        it { does_something }
+      end
+    RUBY
+  end
+
+  it 'registers an offense for multiline blocks without empty line before' do
+    expect_offense(<<-RUBY)
+      RSpec.describe User do
+        before { do_something_else }
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Add an empty line after `before`.
+        before do
+          do_something
+        end
+        ^^^ Add an empty line after `before`.
+        it { does_something }
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      RSpec.describe User do
+        before { do_something_else }
+
+        before do
+          do_something
+        end
+
+        it { does_something }
+      end
+    RUBY
+  end
+
+  it 'allows chained one-liner with different hooks' do
+    expect_no_offenses(<<-RUBY)
+      RSpec.describe User do
+        before { do_something_else }
+        after { do_something_else }
+
+        it { does_something }
+      end
+    RUBY
+  end
+
+  context 'when AllowConsecutiveOneLiners option is false' do
```

Comment:
It appears to me that numerous examples are common for both styles. Can you please put them into `shared_examples`?